ietf-ns-underlay-path.yang

```diff
@@ -0,0 +1,178 @@
+        or without modification, is permitted pursuant to, and
+        subject to the license terms contained in, the Revised
+        BSD License set forth in Section 4.c of the IETF Trust's
+        Legal Provisions Relating to IETF Documents
+        (https://trustee.ietf.org/license-info).
+
+        This version of this YANG module is part of RFC XXXX; see
+        the RFC itself for full legal notices.";
+
+
+     revision 2024-03-27 {
+       description "Initial revision";
+       reference
+         "RFC XXXX: IETF Network Slice Topology YANG Data Model";
+     }
+
+     /*
+      * Groupings
+      */   
+     grouping underlay-path {
+       description
+         "Underlay explicit path within a customer intent
+          topology.";
+		 
+       uses nw:network-ref;
+
+       list path-element {
+        key "index";
+        description
+          "List of path elements.";
+        leaf index {
+          type uint32;
+          description
+            "Index of the hop within the underlay path.";
+        }
+        leaf is-strict-hop {
+	      type boolean;
+	      description
+	        "Indicate whether the hop is strict or loose";
+        }
+        choice type {
+          description
+            "Type of the hop.";
+          case node-hop {
+            leaf node-id {
+              type nw:node-id;
+              description
+                "Node identifier.";
+            }
+          }
+          case link-hop {
+            leaf link-id {
+              type nt:link-id;
+              description
+                "Link identifier.";
+            }
+          }
+          case tp-hop {
+            leaf tp-id {
+              type nt:tp-id;
+              description
+                "Termination Point (TP) identifier.";
+            }
+          }
+        }
+       }
+     }
+
+     /*
+      * Augmented data nodes
+      */
+     augment "/ietf-nss:network-slice-services" +
+             "/ietf-nss:slice-service" +
+             "/ietf-nss:connection-groups" +
+             "/ietf-nss:connection-group" +
+             "/ietf-nss:slo-sle-policy" + 
+             "/ietf-nss:custom" + 
+             "/ietf-nss:service-slo-sle-policy" + 
+             "/ietf-nss:sle-policy" +
+             "/ietf-nss:path-constraints" {
+       description
+         "Underlay path for connection group.";
+
+       uses underlay-path;
+     }
+
+     augment "/ietf-nss:network-slice-services" +
+             "/ietf-nss:slice-service" +
+             "/ietf-nss:connection-groups" +
+             "/ietf-nss:connection-group" +
+             "/ietf-nss:connectivity-construct" +
+             "/ietf-nss:slo-sle-policy" + 
+             "/ietf-nss:custom" + 
+             "/ietf-nss:service-slo-sle-policy" + 
+             "/ietf-nss:sle-policy" +
+             "/ietf-nss:path-constraints" {
+       description
+         "Underlay path for connection group.";
+
+       uses underlay-path;
+     }
```

Comment:
These two augment statements have the same description: better to differentiate them